cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -534,6 +535,7 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 				logWithId.Infof("  %s success", stageResult.Type)
 			} else {
 				logWithId.Infof("  %s failure:", stageResult.Type)
+				failedStages = append(failedStages, stageResult.Type)
```

Comment:
I think that the information about the failed stage name should be instead taken from the `osbuildJobResult.OSBuildOutput.Error`, which will contain it if the failure is caused by a stage failure.

See https://github.com/osbuild/osbuild/blob/1b5b015fef9da025ee872b63f46e0dc69f0c108b/osbuild/formats/v2.py#L426-L439

BTW, this also means that all of this is already part of the `osbErrors` added as a detail to `JobError`.

Reply:
Probably it still would make sense to add in the the reason (additionally) as it's shown more prominently in the UI  - but I'll check to use `osbuildJobResult.OSBuildOutput.Error`

Reply:
@thozza at that level I get `osbuildJobResult.OSBuildOutput.Error` only as `json.RawMessage` not sure if I should _try_ to guess the structure of it - or where could I get the type of the structure for sure? (couldn't find something like the `ClientErrorCode` or similar in that context)

![Screenshot_20240502_113019](https://github.com/osbuild/osbuild-composer/assets/8057963/00661b8b-7ac4-48a3-953a-2a7e2292fd6d)

Reply:
I think that a custom unmarshaler would be needed, which would check the value of the `"type"` property and unmarshal to the appropriate struct. We never implemented proper osbuild errors unmarshaling, because it was not needed. Until now.

I agree with putting more information into the "reason" of the error :wink:

Reply:
@thozza ... and I thought that the type is not very strict so it would be more some kind of guessing and trying to cover all types of structs?

Reply:
We control what osbuild returns and it currently returns exactly one type of a struct in case of the stage failure. IMHO, it makes sense to standardize this in osbuild if it is not nice and easy to interpret in osbuild-composer (or any other place that consumes osbuild's JSON output).

We do this type of unmarshaling already in multiple places, e.g.:
https://github.com/osbuild/osbuild-composer/blob/8dece19342d14476a17f2a0a5c9807fed7e24320/internal/weldr/upload.go#L133-L175
https://github.com/osbuild/osbuild-composer/blob/8dece19342d14476a17f2a0a5c9807fed7e24320/internal/target/target.go#L62-L134